### PR TITLE
Feature register implementation

### PR DESCRIPTION
**Pull request by HarishChoure**
Linked Work Items

https://expense-management-system-batch-1-2022.atlassian.net/browse/EMSB12-75
Checklist

 

**Checklist**
- [x] Code adheres to **[Git and Pull Request Guidelines]**
- [x] Code adheres to **[Angular Guidelines]**
- [x] Code adheres to **[Front-end Checklist]**

### CODE REVIEW

**Comment by AshaPatel216** [comment]
Spelling mistake in commit message. Implement **registraion** form -> Implement **registration** form

File: src/app/models/register.model.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+   * User_register interface validate the data type of the user
+   *  */
+export interface User_register{
```
Comment (AshaPatel216) [inline src/app/models/register.model.ts]: Interface name should be in Camel case with first letter uppercase

File: src/app/models/register.model.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+   * User_register interface validate the data type of the user
```
Comment (AshaPatel216) [inline src/app/models/register.model.ts]: Comment format (indentation) is not proper

File: src/app/services/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  providedIn: 'root'
+})
+export class UserService {
+  /**
+   * This is base url link which is common for all
+   *  */
+  private baseUrl = 'https://localhost:44329';
+
+  constructor(private http : HttpClient) { }
+   /**
```
Comment (AshaPatel216) [inline src/app/services/user.service.ts:19]: Add one space after constructor to make code readable

File: src/app/user-sign-up/register/register.component.ts [hunk 4]
```diff
@@ -9,15 +12,18 @@ import { Router } from '@angular/router';
 })
 export class RegisterComponent implements OnInit {
   registerForm!: FormGroup;
-
-  // Constructor
-  constructor(private router: Router){};
-
-  // ngOnInit method
+  /**
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts:15]: Space before constructor. Make this practice to all function too.

File: src/app/user-sign-up/register/register.component.ts [hunk 5]
```diff
@@ -9,15 +12,18 @@ import { Router } from '@angular/router';
...
+   * Constructor
+   */
+  constructor(private add_users : UserService,private router: Router ){};
+  /**
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts:20]: Same as above

File: src/app/user-sign-up/register/register.component.ts [hunk 6]
```diff
@@ -9,15 +12,18 @@ import { Router } from '@angular/router';
...
+   * ngOnInit method
+   * @returns void
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts]: No need to add @returns if it is void

File: src/app/user-sign-up/register/register.component.ts [hunk 7]
```diff
@@ -26,20 +32,77 @@ export class RegisterComponent implements OnInit {
...
 
-  // getter methods
+  /**
+   * name method are used to get name
+   * @retun name
+   *  */
   get name(){
     return this.registerForm.get('name');
   }
+  /**
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts:43]: Space. check this to overall code

File: src/app/user-sign-up/register/register.component.ts [hunk 8]
```diff
@@ -26,20 +32,77 @@ export class RegisterComponent implements OnInit {
...
+   * addUser method are used to subscribe data
+   * @param new_user : User_register
+   *  */ 
+  addUser(new_user: User_register) {
+    this.add_users.addUsers(new_user).subscribe(
+      (x) => {
+        alert("Registration Succesful");
+        this.router.navigate(['/login']);
+      },
+      (error: HttpErrorResponse) => {
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts:71]: Why this error codes are written here?

File: src/app/user-sign-up/register/register.component.ts [hunk 9]
```diff
@@ -26,20 +32,77 @@ export class RegisterComponent implements OnInit {
...
+        else if (error.status === 500) {
+          alert("Server error");
+        }
+        else if (error.status === 501) {
+          alert("Server error");
+        }
+      }
+    );
+  }
+  // Submit method
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts]: remove this duplicate comment

File: src/app/user-sign-up/register/register.component.ts [hunk 10]
```diff
@@ -26,20 +32,77 @@ export class RegisterComponent implements OnInit {
...
+  /**
+   * Submit method
+   * @returns void
+   */
   onSubmit(): void{
+    const formData = this.registerForm.value;
+    const new_user : User_register ={
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts]: Naming convention not followed

File: src/app/user-sign-up/register/register.component.ts [hunk 11]
```diff
@@ -26,20 +32,77 @@ export class RegisterComponent implements OnInit {
...
+     userName : formData.name,
+     emailAddress : formData.email,
+     password : formData.password,
+     appName:'Splitwise',
+   }
+   /**
+   * call add user method
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts]: If you are adding comment inside the method then it should be in '//' or /*---*/ format.

File: src/app/services/user.service.ts [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+  /**
+   * This is base url link which is common for all
+   *  */
+  private baseUrl = 'https://localhost:44329';
+
+  constructor(private http : HttpClient) { }
+
+   /**
+   * addUsers are used to post data in the Api
+   * @retun http response
```
Comment (AshaPatel216) [inline src/app/services/user.service.ts]: Make sure to add description of params & returns not only the return type and the param type.
https://marketplace.visualstudio.com/items?itemName=salbert.comment-ts.
Take care of this from next PR without fail

File: src/app/user-sign-up/register/register.component.ts [hunk 13]
```diff
@@ -9,37 +12,99 @@ import { Router } from '@angular/router';
...
+   *  */
+  get password() {
     return this.registerForm.get('password');
   }
 
-  // Form submit method
-  onSubmit(): void{
+  /**
+   * addUser method are used to subscribe data
+   * @param new_user : UserRegister
```
Comment (AshaPatel216) [inline src/app/user-sign-up/register/register.component.ts]: new_user -> newUser
also the description of params

File: src/app/user-sign-up/register/register.component.ts (hunk 8 above)
Comment (AshaPatel216) [reply src/app/user-sign-up/register/register.component.ts:71]: @HarishChoure waiting for reply here

File: src/app/services/user.service.ts (hunk 12 above)
Comment (HarishChoure) [reply src/app/services/user.service.ts]: OK Dii I will take care of this in the next PR.

File: src/app/user-sign-up/register/register.component.ts (hunk 13 above)
Comment (HarishChoure) [reply src/app/user-sign-up/register/register.component.ts]: > new_user -> newUser also the description of params

Ok Dii

File: src/app/user-sign-up/register/register.component.ts (hunk 8 above)
Comment (HarishChoure) [reply src/app/user-sign-up/register/register.component.ts:71]: Error code are written here for showing alert to the user, If user entering wrong details and If any server error in the backend.

File: src/app/user-sign-up/register/register.component.ts (hunk 11 above)
Comment (HarishChoure) [reply src/app/user-sign-up/register/register.component.ts]: Yes Di, done

**Review by AshaPatel216 — APPROVED** [review]

**Pull request merged**